### DISCUSSION TITLE
WEBMIDI to build a virtual MIDI controller for live DAWs such as Ableton

### DISCUSSION THREAD

**Discussion by PippoApps** [djipco/webmidi#402] [Q&A] [answered]
Hello there,
the question title says it all :)
I am considering building an Electron app to be used as a console to control a DAW for spatialized audio.
Do you think this is possible?
Cheers
P

**Comment by djip-prof** [comment, marked as answer]
It is definitly possible. If you are on Mac, you can use the built-in IAC MIDI driver to route between applications. If you are on Windows, you will need to install a 3rd-party such as [loopMIDI](https://www.tobias-erichsen.de/software/loopmidi.html).

**Reply by PippoApps** [reply]
Thank you! And happy new year :)

**Reply by jonnyhotchkiss** [reply]
> It is definitly possible. If you are on Mac, you can use the built-in IAC MIDI driver to route between applications. If you are on Windows, you will need to install a 3rd-party such as [loopMIDI](https://www.tobias-erichsen.de/software/loopmidi.html).

Hi, is this because web midi can only talk to devices?
Am just wondering if it's theoretically possible to do it all "in the browser", or if that would depend on the browser and platform, if at all possible - eg something like kiwi browser on android, which allows chrome extensions...

Also there's a distinct lack of virtual midi android stuff, so doing it all in webmidi would be great.

Just found your repo whilst looking for webmidi sysex helper stuff... Anything to make getting started and testing/debugging/development a little easier!!

**Reply by djipco** [reply]
The Web MIDI API available in browsers only allows MIDI communication with devices. It is not a system-level MIDI API. It will not let you create virtual MIDI devices. That's just not possible.
